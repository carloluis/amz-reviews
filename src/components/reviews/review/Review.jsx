import React from 'react';
import PropTypes from 'prop-types';
import Info from '../info/Info';
import styles from './Review.scss';

const Review = ({ childAsin, content, created, productTitle, stars, title }) => (
    <div className={styles.container}>
        <div className={styles.header}>
            <i className={styles.image} />
            <Info title="DATE" text={created} />
            <Info title="STARS" text={stars} />
            <Info title={childAsin} text={productTitle} />
        </div>
        <div className={styles.content}>
            <span>{title}</span>
            <p>{content}</p>
        </div>
    </div>
);

Review.propTypes = {
    childAsin: PropTypes.string.isRequired,
    content: PropTypes.string.isRequired,
    created: PropTypes.number.isRequired,
    productTitle: PropTypes.string.isRequired,
    productImg: PropTypes.string.isRequired,
    reviewId: PropTypes.string.isRequired,
    reviewCreated: PropTypes.number.isRequired,
    stars: PropTypes.number.isRequired,
    title: PropTypes.string.isRequired
};

export default Review;
