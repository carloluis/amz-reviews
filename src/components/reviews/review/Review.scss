.container {
    max-width: 750px;
    height: 242px;
    margin: 32px auto;
    border: 1px solid;
    border-radius: 16px;
    padding: 32px;

    .header {
        display: flex;
        justify-content: space-around;
    }

    .content {
        & > span {
            font-weight: 600;
        }
    }
}
