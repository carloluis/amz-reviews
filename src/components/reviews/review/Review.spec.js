import React from 'react';
import renderer from 'react-test-renderer';
import Review from './Review';

const PROPS = {
    childAsin: 'childAsin?',
    content: 'Lorem ipsum',
    created: 1528940971610,
    productTitle: 'Product Title',
    productImg: 'Image',
    reviewId: 'Review Id',
    reviewCreated: 1528940971610,
    stars: 5,
    title: 'Title'
};

describe('<Review/>', () => {
    it('snapshot', () => {
        const result = renderer.create(<Review {...PROPS} />);
        expect(result).toMatchSnapshot();
    });
});
