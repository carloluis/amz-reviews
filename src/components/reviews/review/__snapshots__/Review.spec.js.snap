// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Review/> snapshot 1`] = `
<div
  className="container"
>
  <div
    className="header"
  >
    <i
      className="image"
    />
    <div
      className="container"
    >
      <div
        className="title"
      >
        DATE
      </div>
      <div
        className="text"
      >
        13.6.2018
      </div>
    </div>
    <div
      className="container"
    >
      <div
        className="title"
      >
        STARS
      </div>
      <div
        className="text"
      >
        ★★★★★
      </div>
    </div>
    <div
      className="container"
    >
      <div
        className="title"
      >
        childAsin?
      </div>
      <div
        className="text"
      >
        Product Title
      </div>
    </div>
  </div>
  <div
    className="content"
  >
    <span>
      Title
    </span>
    <p>
      Lorem ipsum
    </p>
  </div>
</div>
`;
